go.mod

```diff
@@ -54,36 +53,31 @@ require (
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
 	github.com/grpc-ecosystem/grpc-gateway v1.12.0
 	github.com/grpc-ecosystem/grpc-opentracing v0.0.0-20171214222146-0e7658f8ee99
-	github.com/hashicorp/go-immutable-radix v1.0.0 // indirect
-	github.com/hashicorp/go-msgpack v0.5.3 // indirect
 	github.com/hashicorp/hcl v0.0.0-20171017181929-23c074d0eceb // indirect
-	github.com/hashicorp/raft v1.0.0 // indirect
 	github.com/iancoleman/strcase v0.0.0-20190422225806-e506e3ef7365 // indirect
 	github.com/jaswdr/faker v1.0.2
 	github.com/jhump/protoreflect v1.5.0
 	github.com/jonboulle/clockwork v0.1.0 // indirect
 	github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
 	github.com/kylelemons/godebug v0.0.0-20170820004349-d65d576e9348
 	github.com/leanovate/gopter v0.2.4
-	github.com/lib/pq v0.0.0-20171126050459-83612a56d3dd
+	github.com/lib/pq v1.2.0
```

Comment:
⚠️ We've seen problems with this upgrade that we haven't been able to dig into for yet: https://github.com/chef/automate/pull/2115

Reply:
@stevendanna so I removed this change to the pq version from this PR, but I am still seeing the same build kite failures in this PR as are occurring for the other (#2115) PR. Unless there's some build cache that's keeping bad builds around, it seems like there might be some other culprit for the failures. But I'm at a loss to explain it.

Reply:
@danielsdeleo It looks like `lib/pq` files are still being updated in this PR.

Reply:
😬 indeed. I was looking only at `go.mod`, which I guess I was able to make out-of-sync with the actual content in `vendor/`. Do you have an idea of how we might detect that case? Seems like a thing we could put in the repo health ci job. `go mod verify` doesn't seem to solve the problem.

Reply:
K, so after flailing at this a bit I finally learned that newer versions of nats-streaming-server have lib/pq 1.2.0 in their go.mod file and older versions of nats-streaming-server won't work with the renamed nats.go (formerly called go-nats) library. There are some recent-ish versions of nats-streaming-server that require lib/pq 1.1.1, but that version seems to also have the same behavior as 1.2.0.

Reply:
> go mod verify doesn't seem to solve the problem.

Indeed it does not.  There is an open issue upstream:

https://github.com/golang/go/issues/27348

I think the only alternative is to run `make revendor` and see if there is a diff.  That might be a bit of a bummer as it means re-downloading everything...

> There are some recent-ish versions of nats-streaming-server that require lib/pq 1.1.1, but that version seems to also have the same behavior as 1.2.0.

OK.  We had delayed looking into this, but we can perhaps move that up since it is blocking other work.  My current theory is that the issue is here:

https://github.com/lib/pq/commit/6e2a335bdf9878301a663728f1b8b6483848beed

as I _think_ that results in certain types of errors that per previous retried not being retried now which is hitting us when our pg connections hit the idle timeout in pg-gateway.

Reply:
> go mod verify doesn't seem to solve the problem.

Indeed it does not.  There is an open issue upstream:

https://github.com/golang/go/issues/27348

I think the only alternative is to run `make revendor` and see if there is a diff.  That might be a bit of a bummer as it means re-downloading everything...

> There are some recent-ish versions of nats-streaming-server that require lib/pq 1.1.1, but that version seems to also have the same behavior as 1.2.0.

OK.  We had delayed looking into this, but we can perhaps move that up since it is blocking other work.  My current theory is that the issue is here:

https://github.com/lib/pq/commit/6e2a335bdf9878301a663728f1b8b6483848beed

as I _think_ that results in certain types of errors that per previous retried not being retried now which is hitting us when our pg connections hit the idle timeout in pg-gateway.